.pre-commit-config.yaml

```diff
@@ -1,8 +1,33 @@
 repos:
-  # Autoformat: Shell files
+  # C++
   - repo: local
     hooks:
     - id: format
       name: format
       entry: bash ./scripts/format.sh
       language: system
+      pass_filenames: false
+
+  # Python formatter
+  - repo: https://github.com/psf/black
+    rev: 22.6.0
+    hooks:
+      - id: black
+        language_version: python3
+        args: ['--line-length=105']
+        files: ^(src/bindings/python/|scripts/)
+
+  # Python linter
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 5.0.4
+    hooks:
+      - id: flake8
+        # E501: ignore line length (black auto-formats)
+        # E731: ignore replace-lambda-with-def warnings
+        # E203: ignore "whitespace before ':'"
+        args: [
+          '--max-line-length=105',
+          '--extend-ignore=E501,E731,E203',
+          '--extend-exclude=src/bindings/python/__init__.py'
+        ]
+        files: ^(src/bindings/python/|scripts/)
```

Comment:
is there any chance we can allow the use of single quotes and not force double quotes. for whatever reason i am partial to them, unless you have a strong opinion

Reply:
na, I don't and also don't understand why that's an issue for a formatter. I included it but will wait for the master hot fix at #3714 before pushing